### ISSUE TITLE
Unable to upload

### ISSUE THREAD

**Issue by o7n** [fresns/fresns#39]
- using S3 Storage
- configured Systems - Storage - Image - Service Config
- go to Systems - General - Logo - Upload 
- select file
- press Save

Error **Attempt to read property "fid" on null**
plugins/S3Storage/app/Services/CmdWordService.php:86

`$fileInfo = FileHelper::fresnsFileInfoById($fileModel->fid, $bodyInfo);`

When changing to Local storage plugin the upload does not work, after uploading and saving the "Logo" field becomes blank. No error. There is nothing in storage/app/public/

Permissions are ok
After changing settings I did empty cache

**Comment by jevantang** [comment]
You have only selected the S3 plugin. Have you configured the S3 bucket settings? This error occurred because the upload was unsuccessful, and the file id could not be retrieved.

**Comment by jevantang** [comment]
I just tested two plugins, and both worked without issues.

When using the S3 plugin, you need to configure the bucket information. Set the Filesystem Disk to ‘Remote’.
https://marketplace.fresns.com/open-source/detail/S3Storage

When using the local storage plugin, set the Filesystem Disk to ‘Local’.
https://marketplace.fresns.com/open-source/detail/FileStorage

**Comment by o7n** [comment]
Thank you for the fast response.

Yes, I did all that.

1. S3 - See image - uploading a png gives the error above
2. Local - (Service provider: file storage, filesystem disk: local) - uploading a png does nothing
![fresns-s3](https://github.com/fresns/fresns/assets/19340973/c97abaca-efc6-4d7d-8ca4-19b8673e3887)

Is there any way to troubleshoot, e.g. see some logs or anything like that?

**Comment by jevantang** [comment]
I just posted an article that you can refer to.
https://discuss.fresns.org/post/6hktLCo7

Let’s first confirm whether the file has been uploaded to the S3 bucket. After that, we can investigate any access issues. If the bucket is private, you will need to enable the temporary link feature.

**Comment by jevantang** [comment]
Please access the bucket and manually check if the file has been uploaded.

The plugin is functioning correctly. If the file is not in the storage bucket, it indicates there was an issue during the upload process, and the S3 bucket configuration needs to be investigated.

**Comment by o7n** [comment]
Thank you for that article! It confirms that I have set things up correctly...
The file is not in the bucket however. What would be the best way to troubleshoot?

When using `aws cp test.txt s3://bucketname` from my Linux command line, using the same access key ID and secret, I am able to upload files.

**Comment by jevantang** [comment]
![WX20240701-221957@2x](https://github.com/fresns/fresns/assets/22316055/712d7c97-4b50-4aa7-9a15-61c3c2b2d20b)

If you’re not familiar with coding, the only options are to check the S3 configuration and clear the cache.

I just conducted a test using Cloudflare R2, an object storage service that supports the S3 protocol, and successfully uploaded files using the S3 storage plugin.

**Comment by o7n** [comment]
I am familiar with coding with 35 years of experience

Going to set up a R2 test now, will report back.

**Comment by jevantang** [comment]
https://github.com/fresns/fresns/blob/3.x/app/Utilities/FileUtility.php#L61-L65

The upload process occurs here, without any custom code involved; it is entirely handled by the storage system provided by the Laravel framework.

**Comment by o7n** [comment]
Thank you for your directions. I have debugged this and when setting `throw` to `true` I was able to get

`
   League\Flysystem\UnableToWriteFile  Unable to write file at location: xyz.txt. Error executing "PutObject" on "https://bucketname.bucketname.s3.us-east-1.amazonaws.com/xyz.txt"; AWS HTTP error: cURL error 60: SSL: no alternative certificate subject name matches target host name 'bucketname.bucketname.s3.us-east-1.amazonaws.com' (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://bucketname.bucketname.s3.us-east-1.amazonaws.com/xyz.txt.
`

Note how the bucketname appears twice and this was happening because I did not realize that the endpoint should just be `s3.us-east-1.amazonaws.com` without the bucket name.

Emptying the "Bucket Endpoint" setting resolved the issue.
It would be good to set "throw" to true thus enabling the actual error message to surface.

Thank you for your help!
